code/modules/projectiles/updated_projectiles/guns/energy.dm

```diff
@@ -371,3 +371,236 @@
+	if(!magazine.default_ammo)
+		log_debug("ERROR CODE A1: null ammo while reloading. User: <b>[user]</b>")
+		ammo = ammo_list[/datum/ammo/bullet] //Looks like we're defaulting it.
+	else
+		ammo = ammo_list[overcharge? magazine.overcharge_ammo : magazine.default_ammo]
+		//to_chat(user, "DEBUG: REPLACE AMMO. Ammo: [ammo]")
+
+
+/obj/item/weapon/gun/energy/lasgun/load_into_chamber(mob/user, overcharge_check = FALSE)
+	//The workhorse of the bullet procs.
+ 	//If we have a round chambered and no active attachable, we're good to go.
+	if(in_chamber && !active_attachable)
+		if(overcharge_check) //Check to see if we have the proper ammo in chamber to match the overcharge fire mode
+			var/obj/item/projectile/reg_ammo = rnew(/obj/item/projectile, src)
+			reg_ammo.generate_bullet(ammo_list[current_mag.default_ammo])
+			var/obj/item/projectile/over_ammo = rnew(/obj/item/projectile, src)
+			over_ammo.generate_bullet(ammo_list[current_mag.overcharge_ammo])
+			//to_chat(user, "<span class='warning'>DEBUG: Load Into Chamber. Overcharge: [overcharge] Chamber: [in_chamber] Reg_Ammo: [reg_ammo] Over_Ammo: [over_ammo] </span>")
+			if(overcharge && in_chamber.name == reg_ammo.name)
+				//to_chat(user, "<span class='warning'>DEBUG: Chamber ammo replaced with overcharge. Chamber: [in_chamber] </span>")
+				in_chamber = null //clean the chamber of the erroneous round
+				current_mag.current_rounds += 1 //refund cost of a standard shot.
+				return ready_in_chamber(user, 1)
+			else if (!overcharge && in_chamber.name == over_ammo.name)
+				//to_chat(user, "<span class='warning'>DEBUG: Chamber ammo replaced with regular shot. Chamber: [in_chamber] </span>")
+				in_chamber = null //clean the chamber of the erroneous round
+				current_mag.current_rounds += OVERCHARGE_AMMO_COST //refund cost of an overcharge shot.
+				return ready_in_chamber(user, 2)
+		return in_chamber //Already set!
+	//Let's check on the active attachable. It loads ammo on the go, so it never chambers anything
+	if(active_attachable)
+		if(active_attachable.current_rounds > 0) //If it's still got ammo and stuff.
+			active_attachable.current_rounds--
+			return create_bullet(active_attachable.ammo)
+		else
+			to_chat(user, "<span class='warning'>[active_attachable] is empty!</span>")
+			to_chat(user, "<span class='notice'>You disable [active_attachable].</span>")
+			playsound(user, active_attachable.activation_sound, 15, 1)
+			active_attachable.activate_attachment(src, null, TRUE)
+	else
+		return ready_in_chamber(user)//We're not using the active attachable, we must use the active mag if there is one.
+
+
+/obj/item/weapon/gun/energy/lasgun/ready_in_chamber(mob/user, switch_modes = FALSE)
+	if(current_mag && current_mag.current_rounds > 0)
+		if(current_mag.current_rounds < ammo_per_shot && overcharge)
+			if(istype(src, /obj/item/weapon/gun/energy/lasgun))
+				var/obj/item/weapon/gun/energy/lasgun/L = src
```

Comment:
...Why are you checking that src is a lasgun inside a redefined proc for the lasgun